app/Http/Controllers/ArticuloController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Articulo;
+
+class ArticuloController extends Controller
+{
+    public function index(Request $request)
+    {
+        if (!$request->ajax()) return redirect('/');
+
+        $buscar = $request -> buscar;
+        $criterio = $request -> criterio;
+
+        if ($buscar == ''){
+            $articulos = Articulo::join('categorias', 'articulos.idcategoria', '=', 'categorias.id')
+                ->select('articulos.id', 'articulos.idcategoria', 'articulos.codigo', 'articulos.nombre', 'categorias.nombre as nombre_categoria', 'articulos.precio_venta', 'articulos.stock', 'articulos.descripcion', 'articulos.condicion')
+                ->orderBy('articulos.id', 'desc') -> paginate(5);
+        }
+        else {
+            $articulos = Articulo::join('categorias', 'articulos.idcategoria', '=', 'categorias.id')
+                ->select('articulos.id', 'articulos.idcategoria', 'articulos.codigo', 'articulos.nombre', 'categorias.nombre as nombre_categoria', 'articulos.precio_venta', 'articulos.stock', 'articulos.descripcion', 'articulos.condicion')
+                ->where('articulos.'.$criterio, 'like', '%'. $buscar . '%')
+                ->orderBy('articulos.id', 'desc') -> paginate(5);
+        }
+
+
+        return [
+            'pagination' => [
+                'total'        => $articulos-> total(),
+                'current_page' => $articulos-> currentPage(),
+                'per_page'     => $articulos-> perPage(),
+                'last_page'    => $articulos-> lastPage(),
+                'from'         => $articulos-> firstItem(),
+                'to'           => $articulos-> lastItem(),
+            ],
+            'articulos' =>$articulos
+        ];
+
+    }
+
+    public function store(Request $request)
```

Comment:
Please use fill function to store article

Reply:
Done